src/main/java/com/gamesage/store/controller/CustomLogoutHandler.java

```
@@ -0,0 +1,40 @@
+package com.gamesage.store.controller;
+
+import com.gamesage.store.domain.model.AuthToken;
+import com.gamesage.store.security.auth.HeaderName;
+import com.gamesage.store.security.service.AuthService;
+import com.gamesage.store.service.TokenService;
+import com.gamesage.store.util.TokenParser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.web.authentication.logout.LogoutHandler;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Objects;
+
+public class CustomLogoutHandler implements LogoutHandler {
+
+        @Autowired
+        private AuthService authService;
+        @Autowired
+        private TokenService tokenService;
+
+        @Override
+        public void logout(HttpServletRequest request, HttpServletResponse response, Authentication authentication) {
+                String tokenFromHeader = request.getHeader(HeaderName.TOKEN_HEADER);
```

Don't silently drop if there's no token, an exception should be thrown